### PR TITLE
Fade out unnecessary usings

### PR DESCRIPTION
**Pull request by akshita31**
Related issue: https://github.com/OmniSharp/omnisharp-vscode/issues/1179 
https://github.com/OmniSharp/omnisharp-vscode/issues/1245

Related API: https://github.com/Microsoft/vscode/issues/20219
https://code.visualstudio.com/updates/v1_25#_diagnostictag

Fade the unnecessary usings
usings

### CODE REVIEW

File: src/features/diagnosticsProvider.ts [hunk 1]
```diff
@@ -325,7 +325,14 @@ class DiagnosticsProvider extends AbstractSupport {
     private static _asDiagnostic(quickFix: protocol.QuickFix): vscode.Diagnostic {
         let severity = DiagnosticsProvider._asDiagnosticSeverity(quickFix.LogLevel);
         let message = `${quickFix.Text} [${quickFix.Projects.map(n => DiagnosticsProvider._asProjectLabel(n)).join(', ')}]`;
-        return new vscode.Diagnostic(toRange(quickFix), message, severity);
+        let diagnostic =  new vscode.Diagnostic(toRange(quickFix), message, severity);
+        if(diagnostic.message.includes("Unnecessary using directive"))
```
[the remark this answers is not in the document]
Comment (akshita31) [reply src/features/diagnosticsProvider.ts]: Yes that sounds right. Will look into it.

**Comment by nitin88** [comment]
Any update on this?

**Comment by savpek** [comment]
I played this feature little:

image

Theres Descriptor.CustomTags field on diagnostic that has value 'Unnecessary' when diagnostic should be faded out.

Updated omnishar-roslyn code (locally) with https://github.com/savpek/omnisharp-roslyn/commit/a29348c293e3c71f0bc679c98c25067a85842a51

However CS0xxx diagnostics don't have that tag field, it is included only on roslyn analysis diagnostics (dunno why?). It seems very common that theres specific 'hidden' diagnostics with 'unnesessary' tag which only purpose is to fadeout parts of code. Like that IDE0035 in image.

In image theres hard coded CS0xxx codes to only fadeout items (CS0162, CS8019) and not show squiggle under text. They are unnecessary usings and unreachable code. I liked that look much better 😄 There may be better generic way to solve this, since there's IDE analysis that tells to hide those lines. However CS diagnostic adds squiggles since it don't have tags but i don't know how to solve that which IDE analysis is meant to hide certain CS analysis in which locations.

**Comment by Ciberusps** [comment]
Updates?

**Comment by akshita31** [comment]
@Ciberusps The relevant changes are already in OmniSharp and we will be working on consuming the new omnisharp version. It should be available soon.

**Comment by akshita31** [comment]
I will close this PR in favor of #2873

**Pull request closed without merging**